tests/ssl-opt.sh

```diff
@@ -2107,23 +2112,23 @@ run_test    "key size: TLS-ECDHE-ECDSA-WITH-AES-128-CCM-8" \
             -c "Key size is 128"
 
 requires_config_enabled MBEDTLS_X509_CRT_PARSE_C
-requires_config_enabled MBEDTLS_MD_CAN_MD5
 # server5.key.enc is in PEM format and AES-256-CBC crypted. Unfortunately PEM
 # module does not support PSA dispatching so we need builtin support.
 requires_config_enabled MBEDTLS_CIPHER_MODE_CBC
-requires_config_enabled MBEDTLS_AES_C
+requires_cipher_enabled AES
```

Comment:
The test case is still not executed, neither in the default configuration nor any other. It turns out that `requires_cipher_enabled` doesn't currently support AES-CBC. You can see what's going on by enabling tracing (`set -x` before this stanza, `set +x` after) and looking in the trace where `SKIP_NEXT` is set to `YES`.

But, anyway, as indicated in the comment, we need the _built-in_ AES-CBC here, because the decryption of the key relies on code in the PEM module that does not support PSA. So `requires_config_enabled MBEDTLS_AES_C` needs to stay (and so does `requires_config_enabled MBEDTLS_CIPHER_MODE_CBC`).

Reply:
The CI won't pass while `MBEDTLS_AES_C` is required because the following driver analysis is failing:
```
19:16:15 *** Analyze driver test_psa_crypto_config_accel_cipher_aead_cmac vs reference test_psa_crypto_config_reference_cipher_aead_cmac ***
19:16:15  
19:16:15  Error: PASS -> SKIP/FAIL: ssl-opt;TLS: password protected server key
19:16:15  Error: PASS -> SKIP/FAIL: ssl-opt;TLS: password protected client key
19:16:15  Error: PASS -> SKIP/FAIL: ssl-opt;TLS: password protected server key, two certificates
```

because `component_test_psa_crypto_config_accel_cipher_aead_cmac` unsets `MBEDTLS_AES_C` and `MBEDTLS_CIPHER_MODE_CBC`, whereas `common_psa_crypto_config_accel_cipher_aead_cmac` uses config `helper_libtestdriver1_adjust_config "full"`, which does not unset these.

Is the solution to unset one of these ciphers in `common_psa_crypto_config_accel_cipher_aead_cmac` to match the expected behaviour?

Reply:
In `tests/scripts/analyze_outcomes.py`, there's a list of exceptions for driver parity. One of the exceptions for `analyze_driver_vs_reference_cipher_aead_cmac` is
```
                # Encrypted keys are not supported so far.
                # pylint: disable=line-too-long
                'test_suite_pkparse': [
                    'Key ASN1 (Encrypted key PKCS12, trailing garbage data)',
                    'Key ASN1 (Encrypted key PKCS5, trailing garbage data)',
                    re.compile(r'Parse (RSA|EC) Key .*\(.* ([Ee]ncrypted|password).*\)'),
```

For the same reason, the `password protected` test cases in `ssl-opt.sh` should be listed as an exception.